### PR TITLE
feat(network): integrate Fuse Network

### PR DESCRIPTION
**Pull request by developerfred**
## Description

Layer2 Age: Add Fuse Network on MaskBook

### **Dexs**

- [x] [FuseFi](https://app.fuse.fi/#/swap)  
- [x] [ELK Finance](https://elk.finance/) 

### Contracts
- [x] [BalanceChecker](https://explorer.fuse.io/address/0x2690906609A9AF3bF08B8384CD9A0F97A88A7E9c/contracts) 
- [x] ITO 0x2F4aFCcE26e44b8DaAf7074800A21b7d17Fd8EEb
- [x]  MaskBox [0x97369fEE7db34E0BfE47861f2ec44b4378d13eB4](https://explorer.fuse.io/address/0x97369fee7db34e0bfe47861f2ec44b4378d13eb4)
- [x]  NFTRedPacket [0xdB93cCd481012bB5D1E2c8d0aF7C5f2940c00fdC](https://explorer.fuse.io/address/0xdB93cCd481012bB5D1E2c8d0aF7C5f2940c00fdC)
- [x]  RedPacket [0x2F4aFCcE26e44b8DaAf7074800A21b7d17Fd8EEb](https://explorer.fuse.io/address/0x2F4aFCcE26e44b8DaAf7074800A21b7d17Fd8EEb)

### Sourcify
- [ ] https://github.com/ethereum/sourcify/pull/621

### External Data
- [x] Coingecko
- [x] Coinmarketcap
- [ ] DeBank **Fuse Netowork is not yet listed** [check](https://openapi.debank.com/docs) [proposal](https://debank.com/vote/1663)

### Mask Assets 
- [x] https://github.com/DimensionDev/assets/pull/4

### CODE REVIEW

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

**Review by Jack-Works — DISMISSED** [review]

File: packages/web3-constants/evm/mask-box.json [hunk 1]
```diff
@@ -12,6 +12,8 @@
         "Arbitrum": "",
         "Arbitrum_Rinkeby": "",
         "xDai": "",
+        "Fuse": "0x97369fEE7db34E0BfE47861f2ec44b4378d13eB4",
```
Comment (UncleBill) [inline packages/web3-constants/evm/mask-box.json]: We don't have MaskBox contract deployed on Fuse yet. Please remove it.

You should not add contracts deployed by youself.

File: packages/web3-constants/evm/nft-red-packet.json [hunk 2]
```diff
@@ -12,6 +12,8 @@
         "Arbitrum": "",
         "Arbitrum_Rinkeby": "",
         "xDai": "",
+        "Fuse": "0xdB93cCd481012bB5D1E2c8d0aF7C5f2940c00fdC",
```
Comment (UncleBill) [inline packages/web3-constants/evm/nft-red-packet.json]: Please check out https://github.com/DimensionDev/RedPacket.

File: packages/web3-constants/evm/red-packet.json [hunk 3]
```diff
@@ -92,6 +100,8 @@
         "Arbitrum": 909051,
         "Arbitrum_Rinkeby": 0,
         "xDai": 18101937,
+        "Fuse": 14826468,
```
Comment (UncleBill) [inline packages/web3-constants/evm/red-packet.json:103]:
```suggestion
        "Fuse": 14952062,
```
https://explorer.fuse.io/tx/0x392c5a78c2e7cfddd495f3858fb7207b4b88fb4c4a360f0077a9794c69c36c72/internal-transactions
image

File: packages/web3-constants/evm/token-asset-base-url.json [hunk 4]
```diff
@@ -39,6 +39,11 @@
         "Arbitrum": [],
         "Arbitrum_Rinkeby": [],
         "xDai": [],
+        "Fuse": [
+            "https://raw.githubusercontent.com/dimensiondev/assets/master/blockchains/fuse",
+            "https://rawcdn.githack.com/dimensiondev/assets/master/blockchains/fuse"
```
Comment (UncleBill) [inline packages/web3-constants/evm/token-asset-base-url.json:44]: We don't have assets for Fuse in https://github.com/DimensionDev/assets/tree/master/blockchains yet. Could you help to collect?

Comment (developerfred) [reply packages/web3-constants/evm/token-asset-base-url.json:44]: Sure!

**Comment by guanbinrui** [comment]
I'm closing this PR. Cause it doesn't belong to any bounty issue.

**Pull request closed without merging**